plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,40 +177,74 @@ export class WinstonInstrumentation extends InstrumentationBase {
       ) {
-        const span = trace.getSpan(context.active());
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
+              record = injectRecord(spanContext, record);
+              instrumentation._callHook(span, record);
+              // Inject in metadata argument
+              let isDataInjected = false;
+              for (let i = args.length - 1; i >= 0; i--) {
+                if (typeof args[i] === 'object') {
+                  args[i] = Object.assign(args[i], record);
+                  isDataInjected = true;
+                  break;
+                }
+              }
+              if (!isDataInjected) {
+                const insertAt =
+                  typeof args[args.length - 1] === 'function'
+                    ? args.length - 1
+                    : args.length;
 
-        if (!span) {
-          return original.apply(this, args);
+                args.splice(insertAt, 0, record);
+              }
+            }
+          }
         }
 
-        const spanContext = span.spanContext();
+        return original.apply(this, args);
+      };
+    };
+  }
 
-        if (!isSpanContextValid(spanContext)) {
-          return original.apply(this, args);
-        }
+  private _getPatchedConfigure() {
+    return (original: Winston3ConfigureMethod) => {
+      const instrumentation = this;
+      return function patchedConfigure(
+        this: never,
+        ...args: Parameters<typeof original>
+      ) {
 
-        for (let i = args.length - 1; i >= 0; i--) {
-          if (typeof args[i] === 'object') {
-            const record = args[i];
-            injectRecord(spanContext, record);
-            instrumentation._callHook(span, record);
-            return original.apply(this, args);
+        const config = instrumentation.getConfig();
+        if (!config.disableLogSending) {
+          if (args && args.length > 0) {
+            // Try to load Winston transport
+            try {
+              const { OpenTelemetryTransport } = require("@opentelemetry/winston-transport");
+              const originalTransports = args[0].transports;
+              let newTransports = Array.isArray(originalTransports)
+                ? originalTransports
+                : [];
+              const openTelemetryTransport = new OpenTelemetryTransport();
+              if (originalTransports && !Array.isArray(originalTransports)) {
+                newTransports = [originalTransports];
+              }
+              newTransports.push(openTelemetryTransport);
+              args[0].transports = newTransports;
+            }
+            catch (err) {
+              instrumentation._diag.error('OpenTelemetry Winston transport is not available', err);
```

Comment:
Not sure if an error log is a good choice here.
The default is `disableLogSending: false` but on default `@opentelemetry/winston-transport` is not installed.
So on default an error log will be created to my understanding.

Maybe change `disableLogSending` to `enableLogSending` (with default false)?
Or use 3 states:
* `enabled` - error log if `@opentelemetry/winston-transport` missing
* `disabled` - never try to use `@opentelemetry/winston-transport`
* `auto`- try to load `@opentelemetry/winston-transport` and use it if present but ignore if not (maybe info/debug log)

Reply:
My only problem changing disableLogSending to enableLogSending is that it will not align with Bunyan instrumentation anymore, changing default is also problematic, I would expect users want this feature to be on by default, maybe changing the error to a info kind of message would be enough. I would argue this diagnostic error or info message is actually useful for customers so they understand something is missing.

Reply:
I'm not sure if everyone really uses an OTLP endpoint for logs. There are existing tools which do log collection even for services which are not instrumented by OpenTelemetry so why should these users switch to direct sending and potentially get duplicate data?

Anyhow, I agree that we should stay consistent. So either establish a new standard and adapt bunyan instrumentation or define that bunyan is the standard and apply it to others.

Personally I would be even in favor that this is synced across techs but I doubt we will get this into some sort of generic spec in a reasonable timeframe.

Reply:
Ok, changed to enableLogSending with default false, it would be good to change Bunyan to align with this in the future

Reply:
Thought once more about this and I think for the first step it's likely better to stay in sync with bunyan and go for enabled on default.
I forgot that the actual decision of exporting logs is done by configuring SDK and exporter,

Sorry for the back and forth.